packages/oidc-server/src/contractUpdater.ts

```diff
@@ -29,19 +29,27 @@ export class ContractUpdater {
     );
   }
 
+  private addIssHashToKeys(keys: Key[], issHash: string): Key[] {
```

Comment:
The typing for this function is weird. If you are adding `issHash` into key, that means that whatever was in `keys` wasn't actually a key.

I believe this compiles because of how `getNewKeys` works. It has this line inside:

`return results.filter((key): key is Key => key !== null);`

I don't love this because this is hinting that what this function returns is a `Key` but it's actually missing one attribute. Do you believe we can make the typing more explicit?

The simplest option is maybe add the `issHash` directly in `getNewKeys`.